File: Ml/database/d4.py
import pymysql

connection = None
# 아이디가 guest인 데이터만 가져와라
try:
  connection = pymysql.connect(host='localhost',
                              user='root',
                              password='12341234',
                              db='python_db',
                              charset='utf8mb4'                              
                              ,cursorclass=pymysql.cursors.DictCursor)
  with connection.cursor() as cursor:     
    # 파라미터를 넣을때는 '값' => %s로 치환(홀따움표 포함)
    sql = '''
    select src, slang as label from tbl_trans_log
    where uid=%s;
    '''
    # 파리미터는 쿼리 수행시 튜플로 전달(2번인자)
    cursor.execute(sql, ('guest',))
    result = cursor.fetchall() 
    print(result)    
except Exception as e:
  print('오류발생', e)
finally:
  if connection:
    connection.close()

File: Another_Web/node_basic/database/d4.js
const mysql = require('mysql')

var connection = mysql.createConnection({
    host     : 'localhost',
    user     : 'root',
    password : '12341234',
    database : 'nodedb'
});
connection.connect( (err)=>{
    if( err ){
        console.log('접속 오류', err)
    }else{   
        // 아이디와 비번을 외부에서 세팅되게 수정 => 일반화(누구나 사용가능)     
        let sql = `select * from tbl_users where
                        uid = ? and upw = ? ;`
        // query() 함수의 2번인자가 sql 문자열에 동적으로 데이터를 세팅하는 위치
        // 2번인자는 생략가능
        connection.query( sql, ['guest', '1'], (error, rows)=>{
            console.log( error, rows[0].name )            
            connection.end( ( e )=>{
                console.log('접속 종료?', e)
            } );
        } ) 
    }
} );
// d5.js 생성
// 해당 코드를 누구나 불러서 사용할수 있게 함수의 모듈화를 구성